module/User/src/Service/ApiPrincipalService.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace User\Service;
+
+use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
+use User\Form\ApiPrincipal as ApiPrincipalForm;
+use User\Mapper\ApiPrincipalMapper;
+use User\Model\ApiPrincipal as ApiPrincipalModel;
+use User\Model\Enums\ApiPermissions;
+
+use function array_map;
+
+class ApiPrincipalService
+{
+    public function __construct(
+        protected readonly ApiPrincipalMapper $mapper,
+        protected readonly ApiPrincipalForm $apiPrincipalForm,
+    ) {
```

Comment:
Can you reorder these to follow `Form` -> `Mapper` -> `Service`.

We generally order it to be `Acl` -> `MvcTranslator` -> `Form`s -> `Mapper`s -> `Service`s -> configs.

Reply:
95c03ca44a090e19d4a3bf96aece47686d88037e